Swaps sleep() to use SECONDS define

san7890 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Hey there,

I was shocked by just how deep the deciseconds devil was in the codebase recently, so I decided to go through and run the epic `sleep\((..)\)` regex and mass find-and-replace everything. I think it's swag.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Better readability, better maintainability (probably). Looks nicer too, I don't like thinking in deciseconds for too long. We already pretty much mandate new contributors use SECONDS defines, so let's make sure all the old code is up to snuff as well.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

Nothing that players should have cause for alarm about.

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

san7890 committed:
Swaps sleep() to use SECONDS define

Hey there,

I was shocked by just how deep the deciseconds devil was in the codebase recently, so I decided to go through and run the epic `sleep\((..)\)` and mass find-and-replace everything. I think it's swag.

Fikou reviewed (changes requested):
i think you should audit sleep(1), some of them sleep just to wait a single tick because otherwise stuff would break, i don't know if seconds defines are appropriate there

MrStonedOne commented:
> i think you should audit sleep(1), some of them sleep just to wait a single tick because otherwise stuff would break, i don't know if seconds defines are appropriate there

Correct, such things should use TICKS instead of seconds.

san7890 committed:
SECONDS -> TICKS

I changed anything that seemed reliant on actually waiting a tick here, four instances of sleep(0.1 SECONDS) remain because they seem to be just small breaks to allow some sort of in-game animation?

san7890 commented:
I swapped over everything that seemed dependent on waiting an actual tick to the TICKS define, while stuff that seemed to just be small breaks in an animation sequence were kept as seconds. Let me know if my judgement was incorrect.

MrStonedOne commented:
> Let me know if my judgement was incorrect.

nobody is gonna review this in full like that now, we already did it once and made our mark.

LemonInTheDark commented on code/controllers/master.dm:
```diff
@@ -268,7 +268,7 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 	if(sleep_offline_after_initializations)
 		world.sleep_offline = TRUE
-	sleep(1)
+	sleep(1 TICKS)
```
this changes behavior. Not sure if it should or not but it does.
1 tick with our tick lag is 0.5 seconds. sleep(1) is therefor 2 ticks.

LemonInTheDark replied:
@MrStonedOne advise ya boomer

MrStonedOne replied:
This sleep is correctly changed. the goal is to make byond sleep because if nobody is connected it won't wake up until somebody connects, the time is irrelevant other then 0 will sometimes not sleep so it can't be that.

LemonInTheDark reviewed (changes requested):
Worried. See review.

Using seconds in the MC feels wrong, not sure why

ShizCalev commented on code/controllers/master.dm:
```diff
@@ -268,7 +268,7 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 	if(sleep_offline_after_initializations)
 		world.sleep_offline = TRUE
-	sleep(1)
+	sleep(1 TICKS)
```
`Note: sleep time is in 1/10s units, not server ticks. If your world.tick_lag or world.fps value is different from the default, sleep(1) still means "sleep for 1/10s". To sleep for exactly N ticks, call sleep(N * world.tick_lag).`

`world.tick_lag is 10 / world.fps and vice-versa, so a tick_lag of 0.25 is equal to 40 fps.`

Are you certain changing sleeps in the MC from 1ds to 0.5ds (our servers are set to 20fps, so 10/20) isn't gonna affect anything?

ShizCalev commented on code/game/machinery/doors/door.dm:
```diff
@@ -402,7 +402,7 @@
 /obj/machinery/door/proc/CheckForMobs()
 	if(locate(/mob/living) in get_turf(src))
-		sleep(1)
+		sleep(1 TICKS)
```
are you sure this wasn't for an animation? everything that uses is has animation sleeps.

san7890 replied:
i didn't pay the closest of attention to what it called right afterwards.

san7890 replied:
i can't be certain but i'm going to default to using seconds here because everything in the proc called right after it is in seconds

ShizCalev commented on code/modules/antagonists/cult/runes.dm:
```diff
@@ -875,7 +875,7 @@ structure_check() searches for nearby cultist structures required for the invoca
 				break
 			user.apply_damage(0.1, BRUTE)
-			sleep(1)
+			sleep(1 TICKS)
```
uhm

ShizCalev replied:
sleep(1) = wait `1 ds`
sleep(1 TICKS) = `1 * (10/world.fps)`
world.fps on our servers is 20
ergo this went from applying damage to the person every 1 ds to every 0.5ds.

this is absolutely a behavior change.

LemonInTheDark replied:
Hey that's my line

ShizCalev replied:
> Hey that's my line

Hah, was reviewing all the changes before submitting. Glad we both caught it at least. :P

san7890 replied:
i blanked the fact that it was a while loop

Mothblocks commented:
CC @MrStonedOne on TICKS comments

optimumtact commented:
there is no way you can do this in bulk and not fuck up

san7890 commented:
> there is no way you can do this in bulk and not fuck up

it was an inherent risk

i'll test-merge this a bit to ensure stuff doesn't blow up in the coming days

LemonInTheDark commented:
please do a pass on all the non "sleep for as short as we can" uses of TICKS so they properly match the delay.
Players won't notice, but I will

san7890 committed:
Pass on decimals and three-number long sleeps, tick touchup

san7890 committed:
I FORGOT THE ZERO GOD DAMMIT

waaargh

san7890 committed:
reverts anything that wasn't directly associated to a minimal sleep

this means that everything that wasn't the TGS API, some TGUI function, and the MC is going back to using seconds to ensure we don't effectively halve out some stuff.

san7890 commented:
The MC Loop and the TGUI_Panel both declare they just want a minimal sleep, so those have TICKS instead of SECONDS. The other remaining TICKS are in some TGS API procs, IsBanned(), and View Variables (which appeared to aim for minimal sleeps). I'll get this test merged somewhere and watch it closely later tonight.

Everything else will use seconds to ensure that no player-facing behavior changes in a drastic manner.

optimumtact commented on code/game/machinery/computer/arcade/orion.dm:
```diff
@@ -540,7 +540,7 @@ GLOBAL_LIST_INIT(orion_events, generate_orion_events())
 	say("Oh, God! Code Eight! CODE EIGHT! IT'S GONNA BL-")
 	playsound(loc, 'sound/machines/buzz-sigh.ogg', 25, TRUE)
-	sleep(3.6)
+	sleep(.36 SECONDS)
```
bro stop

optimumtact replied:
literally this is a compile issue

san7890 replied:
This was fixed in 062744d78a6cf80be4c531e32b8f5956c1fb4310, what commit are you reading?

stylemistake replied:
the most recent one
